src/main/java/com/cloudant/client/org/lightcouch/CouchDbClientBase.java

```diff
@@ -330,64 +273,92 @@ public HttpResponse executeRequest(HttpRequestBase request) {
      * @return {@link Response}
      */
     Response delete(URI uri) {
-        HttpResponse response = null;
-        try {
-            HttpDelete delete = new HttpDelete(uri);
-            response = executeRequest(delete);
-            return getResponse(response, Response.class, getGson());
-        } finally {
-            close(response);
-        }
+        HttpConnection connection = Http.DELETE(uri);
+        return executeRequest(connection, Response.class);
     }
 
-
     /**
      * Performs a HTTP GET request.
      *
-     * @return {@link InputStream}
+     * @return Input stream with response
      */
-    InputStream get(HttpGet httpGet) {
-        HttpResponse response = executeRequest(httpGet);
-        return getStream(response);
+    public InputStream get(URI uri) {
+        HttpConnection httpConnection = Http.GET(uri);
+        return executeRequest(httpConnection);
     }
 
     /**
      * Performs a HTTP GET request.
      *
-     * @return {@link InputStream}
+     * @return Class type of object T (i.e. {@link Response}
      */
-    InputStream get(URI uri) {
-        HttpGet get = new HttpGet(uri);
-        get.addHeader("Accept", "application/json");
-        return get(get);
+    public <T> T get(URI uri, Class<T> classType) {
+        HttpConnection connection = Http.GET(uri);
+        if(classType.isInstance(HttpConnection.class)) {

```

Comment:
What is going on here? This seems very strange as the get/put/post/delete methods would be expected to actually do a request, but this branch appears to convert a URI into a `HttpConnection`, so why not just do `Http.GET(uri)` in the caller?